### PR TITLE
[#800] Add support for ubuntu 24.04 and fedora 40

### LINKED ISSUE #800: Add support of Ubuntu 24.04
# Clarification and motivation

Ubuntu 24.04 was released recently. We should provide packages for it.

# Acceptance criteria

Launchpad PPAs provide packages for Ubuntu 24.04.

### PR DESCRIPTION
**Pull request by krendelhoff2**
## Related issue(s)

Resolves #800

#### Related changes (conditional)

- [ ] I checked whether I should update the [README](/serokell/tezos-packaging/tree/master/README.md)

- [ ] I checked whether native packaging works, i.e. native binary packages
  can be successfully built.

#### Stylistic guide (mandatory)

- [ ] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).

### CODE REVIEW

File: docker/package/model.py [hunk 1]
```diff
@@ -313,7 +313,7 @@ def gen_control_file(self, build_deps, run_deps, ubuntu_version, out):
 Section: utils
 Priority: optional
 Maintainer: {self.meta.maintainer}
-Build-Depends: debhelper (>=9), {"dh-systemd (>= 1.5), " if ubuntu_version != "jammy" else ""} autotools-dev, {str_build_deps}
```
Comment (DMozhevitin) [inline docker/package/model.py:316]: Is it right that we don't need this workaround for `noble`?

File: baking/src/tezos_baking/tezos_setup_wizard.py [hunk 2]
```diff
@@ -881,63 +881,57 @@ def get_current_cycle():
 
         current_cycle = get_current_cycle()
 
-        # TODO remove this check when ParisB protocol is activated at mainnet
-        if adaptive_issuance_launch_cycle is not None and int(current_cycle) >= int(
```
Comment (DMozhevitin) [inline baking/src/tezos_baking/tezos_setup_wizard.py:885]: Nitpick: we don't need to have `adaptive_issuance_launch_cycle` variable and `get_adaptive_issuance_launch_cycle` function anymore

**Review by DMozhevitin — APPROVED** [review]
LGTM 👍 

Have only a minor nitpick and question. Feel free to merge it after addressing this nitpick if answer on my question is "yes"

File: docker/package/model.py (hunk 1 above)
Comment (krendelhoff2) [reply docker/package/model.py:316]: Yes, it is needed only for focal

**Pull request merged**